### PR TITLE
Fixes for 0.9

### PR DESCRIPTION
**Pull request by pakoito**

### CODE REVIEW

File: modules/core/arrow-core-data/src/main/kotlin/arrow/core/Either.kt [hunk 1]
```diff
@@ -367,3 +367,15 @@ fun <A, B> B?.rightIfNotNull(default: () -> A): Either<A, B> = when (this) {
   null -> Either.Left(default())
   else -> Either.Right(this)
 }
+
+/**
+ * Applies the given function `f` if this is a [Left], otherwise returns this if this is a [Right].
+ * This is like `flatMap` for the exception.
+ */
+fun <A, B> EitherOf<A, B>.handleErrorWith(f: (A) -> EitherOf<A, B>): Either<A, B> =
```
Comment (JorgeCastilloPrz) [inline modules/core/arrow-core-data/src/main/kotlin/arrow/core/Either.kt]: back to life

File: modules/docs/arrow-docs/docs/docs/README.md [hunk 2]
```diff
@@ -170,7 +170,7 @@ Add the dependencies that you want to use
 ```
 <dependency>
     <groupId>io.arrow-kt</groupId>
-    <artifactId>arrow-core</artifactId>
+    <artifactId>arrow-core-extensions</artifactId>
```
Comment (JorgeCastilloPrz) [inline modules/docs/arrow-docs/docs/docs/README.md]: wops

File: modules/docs/arrow-docs/docs/docs/quickstart/libraries/README.md [hunk 3]
```diff
@@ -61,14 +61,14 @@ Dependencies: arrow-typeclasses
...
 
-### arrow-instances-(core, data)
+### arrow-(core, extras, effects)-instances
```
Comment (JorgeCastilloPrz) [inline modules/docs/arrow-docs/docs/docs/quickstart/libraries/README.md]: *extensions?

File: modules/docs/arrow-docs/docs/docs/arrow/data/nonemptylist/README.md [hunk 4]
```diff
@@ -13,7 +13,7 @@ video: TC6IzE61OyE
 beginner
 
 `NonEmptyList` is a data type used in __Λrrow__ to model ordered lists that guarantee to have at least one value.
-`NonEmptyList` is available in the `arrow-data` module under the `import arrow.data.NonEmptyList`
+`NonEmptyList` is available in the `arrow-extras` module under the `import arrow.data.NonEmptyList`
```
Comment (JorgeCastilloPrz) [inline modules/docs/arrow-docs/docs/docs/arrow/data/nonemptylist/README.md]: arrow-extras-data or arrow-extras-extensions, no arrow-extras available yet

File: modules/docs/arrow-docs/docs/docs/quickstart/libraries/README.md [hunk 5]
```diff
@@ -49,7 +49,7 @@ Datatypes: [`Const`]({{ '/docs/typeclasses/const/' | relative_url }})
...
 
-### arrow-data
+### arrow-extras
```
Comment (JorgeCastilloPrz) [inline modules/docs/arrow-docs/docs/docs/quickstart/libraries/README.md]: arrow-extras-data

File: modules/docs/arrow-docs/docs/docs/quickstart/libraries/README.md [hunk 6]
```diff
@@ -77,9 +77,9 @@ advanced
 
 Advanced [typeclasses]({{ '/docs/typeclasses/intro' | relative_url }}) to be used in programs using the Tagless-final architecture.
 
-It also includes the instances available for datatypes in both arrow-core and arrow-data
+It also includes the instances available for datatypes in both arrow-core and arrow-extras
```
Comment (JorgeCastilloPrz) [inline modules/docs/arrow-docs/docs/docs/quickstart/libraries/README.md]: Do we wanna refer to them in a generic way here? Those artifacts don't exist with that naming yet

File: modules/docs/arrow-docs/docs/docs/quickstart/libraries/README.md [hunk 7]
```diff
@@ -96,7 +96,7 @@ Optics is the functional way of handling immutable data and collections in a way
 
 It can be used alongside annotation processing to generate [simple DSLs]({{ '/docs/optics/dsl/' | relative_url }}) that read like imperative code.
 
-For all the new typeclasses it also includes the instances available for basic types and datatypes in both arrow-core and arrow-data.
+For all the new typeclasses it also includes the instances available for basic types and datatypes in both arrow-core and arrow-extras.
```
Comment (JorgeCastilloPrz) [inline modules/docs/arrow-docs/docs/docs/quickstart/libraries/README.md]: Same question here :thinkingface:

File: modules/docs/arrow-docs/docs/docs/quickstart/libraries/README.md (hunk 6 above)
Comment (pakoito) [reply modules/docs/arrow-docs/docs/docs/quickstart/libraries/README.md]: Yes, that's my plan

**Pull request merged**